src/auth/local/local.strategy.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable, UnauthorizedException } from "@nestjs/common";
+import { PassportStrategy } from "@nestjs/passport";
+import { Strategy } from "passport-local";
+
+import { AuthService } from "../auth.service";
+import { type JwtUser } from "../jwt/jwt.interface";
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private readonly authService: AuthService) {
+    super({ usernameField: "account" });
+  }
+
+  async validate(account: string, password: string) {
+    const user = await this.authService.validateUser(account, password);
+
+    if (!user) {
+      throw new UnauthorizedException();
```

Comment:
I would use 403 Forbidden instead of 401 Unauthorized.
The 401 means that the user leaks the credential to validate its identity, the user could try again with proper credential.
The 403 means the user provided the credential but not valid, should not try again with same request.

Reply:
But this code shows that the user's account information is wrong, there are two situation.
1. The account is true, but the password is false.
2. The account is false.

401 Unauthorized response should be used for missing or bad authentication.
So I think is 401 Unauthorized instead of 403 Forbidden.

Reply:
I get that.The credential is said like JWT Token. In this way is user given account or password is not correct, not the credential.

I will change to use 403 Forbidden. thanks for the clarification.

Reply:
Revised.
[fix: use forbidden when account information is wrong](https://github.com/Cophr/backend/pull/10/commits/43a0e95a2d52442793d6783bee91fb95690b030c)